pkg/image/api/helper.go

```diff
@@ -896,8 +896,12 @@ func LatestObservedTagGeneration(stream *ImageStream, tag string) int64 {
 }
 
 var (
-	reMajorSemantic = regexp.MustCompile(`^[\d]+$`)
-	reMinorSemantic = regexp.MustCompile(`^[\d]+\.[\d]+$`)
+	reMajorSemantic    = regexp.MustCompile(`^[\d]+$`)
```

Comment:
The goal was to select only a subset - something realistic, but not be slavish on it.  `9-9` doesn't really mean anything, `9.2-5` kind of means something, `9.3.5-6` is generally accepted to mean something.  So only supporting minor with dash is probably ok, but even then, that's not full semantic versioning.  I'd say `x.y-` and `x.y.z-w` are reasonable, the others are not.

Also, sorting order is really undefined on sub tags (rc1 is only coincidentally sorted above alpha1, so frankly i'm a little skeptical that we should let those be prioritized in numeric order).